File: backend/routes/disk.py
"""Disk routes."""
from flask import Blueprint, jsonify, request
from utils.helpers import safe_api, validate_json, require_auth
from core import disk

bp = Blueprint("disk", __name__)


@bp.route("/api/disk/devices")
@safe_api
@require_auth
def devices():
    return jsonify(disk.get_block_devices())


@bp.route("/api/disk/usage")
@safe_api
@require_auth
def usage():
    return jsonify({"usage": disk.get_disk_usage()})


@bp.route("/api/disk/fstab")
@safe_api
@require_auth
def fstab():
    return jsonify({"content": disk.get_fstab()})


@bp.route("/api/disk/fstab/save", methods=["POST"])
@safe_api
@require_auth
@validate_json(["content"])
def fstab_save(data):
    return jsonify({"success": disk.save_fstab(data["content"])})


@bp.route("/api/disk/mount", methods=["POST"])
@safe_api
@require_auth
@validate_json(["device", "mountpoint"])
def mount(data):
    out, code = disk.mount_device(data["device"], data["mountpoint"], data.get("fstype", "auto"))
    return jsonify({"success": code == 0, "message": out})


@bp.route("/api/disk/umount", methods=["POST"])
@safe_api
@require_auth
@validate_json(["target"])
def umount(data):
    ok, msg = disk.umount_device(data["target"])
    return jsonify({"success": ok, "message": msg})

File: frontend/src/api/disk.ts
import { api } from './request'
import type { BlockDevice, TaskResponse } from '@/types/api'

export const diskApi = {
  getDevices: () => api<BlockDevice[]>('/disk/devices'),
  getUsage: () => api<{ usage: string }>('/disk/usage'),
  getFstab: () => api<{ content: string }>('/disk/fstab'),
  saveFstab: (content: string) => api<{ success: boolean }>('/disk/fstab/save', { method: 'POST', body: { content } }),
  mount: (device: string, mountpoint: string, fstype?: string) => api<{ success: boolean; message: string }>('/disk/mount', { method: 'POST', body: { device, mountpoint, fstype } }),
  umount: (target: string) => api<{ success: boolean; message: string }>('/disk/umount', { method: 'POST', body: { target } }),
  // RAID
  getRaidArrays: () => api<{ arrays: any[] }>('/raid/arrays'),
  getRaidDevices: () => api<{ devices: any[] }>('/raid/devices'),
  createRaid: (level: string, devices: string[], name?: string) => api('/raid/create', { method: 'POST', body: { level, devices, name } }),
  manageRaid: (device: string, action: string) => api('/raid/manage', { method: 'POST', body: { device, action } }),
  getRaidDetail: (device: string) => api<{ success: boolean; detail: string }>(`/raid/detail?device=${encodeURIComponent(device)}`),
}
